core/node/state_keeper/src/keeper.rs

```diff
@@ -48,6 +50,44 @@ impl Error {
     }
 }
 
+/// Functionality [`BatchExecutor`] + [`ReadStorageFactory`] with an erased storage type.
+#[async_trait]
+trait ErasedBatchExecutor: fmt::Debug + Send {
+    async fn init_batch(
+        &mut self,
+        l1_batch_env: L1BatchEnv,
+        system_env: SystemEnv,
+        stop_receiver: &watch::Receiver<bool>,
+    ) -> Result<BatchExecutorHandle, Error>;
+}
+
+/// The only [`ErasedBatchExecutor`] implementation.
+#[derive(Debug)]
+struct ErasedBatchExecutorImpl<S> {
+    batch_executor_base: Box<dyn BatchExecutor<S>>,
```

Comment:
Bikeshedding: Since `BatchExecutor` itself is no longer a factory, I guess we no longer need the `_base` postfix.